### ISSUE TITLE
List.GetMembers not retrieving all members ?

### ISSUE THREAD

**Issue by carlos-boronat** [banana-j/bananaj#30]
Hello

We have been using this library to integrate with Mailchimp and we have come across what appears to be a Bug

In **MailchimpList.java** at function **getMembers(int count, int offset)**

In the case that  count value is 0  it takes the value of the **list.getMemberCount** as you can see in
```java
if(count != 0){
	list = new JSONObject(getConnection().do_Get(new URL("https://"+connection.getServer()+".api.mailchimp.com/3.0/lists/"+this.getId()+"/members?count="+count+"&offset="+offset),connection.getApikey()));
} else {
	list = new JSONObject(getConnection().do_Get(new URL("https://"+connection.getServer()+".api.mailchimp.com/3.0/lists/"+this.getId()+"/members?count="+this.getMembercount()+"&offset="+offset),connection.getApikey()));
}
```
However, the value of list.GetmemberCount only includes members in **SUSCRIBED** status, so in case of having members in other status, not all records are retrieved.

In our test case in a list with 5 members:

member1 - SUSCRIBED
member2 - PENDING
member3 - SUSCRIBED
member4 - UNSUBSCRIBED
member5 - UNSUBSCRIBED

The value of list.GetmemberCount is 2 and the function would only return the first 2 records.

An additional question:

Would it be possible to send the status parameter to the API of getMembers in order to filter by the different states?

Thank you very much and greetings

**Comment by alexanderwe** [comment]
Hey @carlos-boronat 👋 , thank you for submitting your issue. I will try to investigate your problem on the upcoming weekend. I am a little bit busy right now, but I will try my best !

**Comment by SadiaWaleem** [comment]
Hi,
I am using this library too. 
When I try to get all members data. I am receiving this error

JSONObject["ADDRESS"] not a string.

can you please look into this?

**Comment by gscriver** [comment]
Fixed in release v0.6.1
